### PR TITLE
Revive the dump_heap primitive, more detailed malloc report.

### PR DESCRIPTION
**Pull request by erikcorry**
This report needs a bit more memory, but shows the fragmentation for each 8-byte memory cell.  For it to work we need to output the base64 data in a single line (recognized by Jaguar).

### CODE REVIEW

File: lib/esp32.toit [hunk 1]
```diff
@@ -137,11 +137,21 @@ rtc_user_bytes -> ByteArray:
...
   send_trace_message report
 
 memory_page_report_ -> ByteArray:
   #primitive.esp32.memory_page_report
+
+/**
+Sends (as a system message) a detailed report over the usage of memory at the OS level.
+*/
+dump_heap -> none:
+  report := dump_heap_ 300
```
Comment (kasperl) [inline lib/esp32.toit:153]: Can we get rid of the versions that write to serial/flash from C code? They are unused?

File: src/heap_report.cc [hunk 2]
```diff
@@ -227,12 +227,12 @@ class SerialFragmentationDumper : public HeapFragmentationDumper {
 };
 
 void dump_heap_fragmentation(output_char_t* output_char_fn) {
-  const char* p = "toit serial decode ";
+  const char* p = "jag decode ";
```
Comment (kasperl) [inline src/heap_report.cc:230]: Is this code used?

File: lib/esp32.toit (hunk 1 above)
Comment (erikcorry) [reply lib/esp32.toit:153]: I want to revive them too.

File: system/containers.toit [hunk 3]
```diff
@@ -350,8 +350,8 @@ trace_using_print message/ByteArray --from=0 --to=message.size:
     end := i >= to - BLOCK_SIZE
     prefix := i == from ? "jag decode " : ""
     base64_text := base64.encode message[i..(end ? to : i + BLOCK_SIZE)]
-    postfix := end ? "" : "\\"
-    print_ "$prefix$base64_text$postfix"
+    postfix := end ? "\n" : ""
+    write_on_stderr_ "$prefix$base64_text$postfix" false
```
Comment (kasperl) [inline system/containers.toit:354]: stdout for now

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Pull request merged**